src/legacy/core_plugins/kibana/public/dashboard/dashboard_state_manager.js

```diff
@@ -219,6 +220,12 @@ export class DashboardStateManager {
       FilterUtils.cleanFiltersForComparison(getFilters(state))
     )) {
       store.dispatch(updateFilters(dashboardFilters));
+    } else {
+      // If the user clickes "refresh" query, it'll get here but without any changes, we still
```

Comment:
What if you added a separate method to `DashboardStateManager` to request reload? Then we can call that directly inside `updateQueryAndFetch`.

Reply:
Can you help me see why you prefer that way?  I tried to take as much code out of the `dashboard_app` file because it is coupled to angular, so any logic that didn't require scope, I would put into `dashboard_state_manager`.  Way back `dashboard_app` was a monolithic file with all logic and everything tightly coupled to angular.  Plus, if we pull out into `dashboard_app`, then to eliminate the double reload, we would have to duplicate the logic for determining if the filters changed and already caused an update in here.  Just don't really see it as an improvement. Thoughts?

Reply:
I understand your point of view, I agree that it'd be nice to remove as much as we can that relies on `$scope` and Angular. My argument is that in my brain "reloading" is meant to be an intentional action you take. In the current implementation it's a side effect that happens any time `applyFilters` is called, but the filters and query haven't changed. While that may currently only happen when the refresh button is clicked, couldn't there be other situations where `applyFilters` is called, and nothing changed, and we don't actually want to reload all of the embeddables?

Reply:
Thanks for the explanation. I don't feel that strongly about this way, and I see your point, I can make the change. Stay tuned!